### PR TITLE
Add omrthread_numa_get_current_node()

### PR DESCRIPTION
**Pull request by dnakamura**
Function returns the numa node number the thread is currently executing
on

Signed-off-by: Devin Nakamura <devinn@ca.ibm.com>

### CODE REVIEW

**Comment by DanHeidinga** [comment]
Related to https://github.com/eclipse/openj9/issues/2813

File: thread/linux/omrthreadnuma.c [hunk 1]
```diff
@@ -546,3 +547,17 @@ dumpNumaInfo() {
...
 }
 #endif
+
+
+uintptr_t
+omrthread_numa_get_current_node(){
+    unsigned node = 0;
+#if OMR_PORT_NUMA_SUPPORT
+    if (0 == syscall(SYS_getcpu, NULL, &node, NULL)) {
+        ++node;
```
Comment (DanHeidinga) [inline thread/linux/omrthreadnuma.c]: This feels like there should be a comment here.  Is the numbering schedule 0 based vs 1 based elsewhere?

File: thread/win32/omrthreadnuma.c [hunk 2]
```diff
@@ -148,3 +148,16 @@ omrthread_numa_get_node_affinity(omrthread_t thread, uintptr_t *numaNodes, uintp
 	*nodeCount = 0;
 	return J9THREAD_NUMA_ERR_AFFINITY_NOT_SUPPORTED;
 }
+
+uintptr_t
+omrthread_numa_get_current_node(){
+    USHORT node = 0;
+
+    PROCESSOR_NUMBER processorNumber;
```
Comment (DanHeidinga) [inline thread/win32/omrthreadnuma.c]: tab vs space

File: thread/win32/omrthreadnuma.c [hunk 3]
```diff
@@ -148,3 +148,16 @@ omrthread_numa_get_node_affinity(omrthread_t thread, uintptr_t *numaNodes, uintp
...
+	GetCurrentProcessorNumberEx(&processorNumber);
+	if (GetNumaProcessorNodeEx(&processorNumber, &node)) {
+		return node + 1;
```
Comment (DanHeidinga) [inline thread/win32/omrthreadnuma.c]: same comment needed here as above

File: include_core/thread_api.h [hunk 4]
```diff
@@ -1411,6 +1411,13 @@ omrthread_numa_get_node_affinity(omrthread_t thread, uintptr_t *numaNodes, uintp
 void
 omrthread_numa_set_enabled(BOOLEAN enabled);
 
+
+/**
+ * Gets the NUMA node the caller is being executed on
```
Comment (DanHeidinga) [inline include_core/thread_api.h]: Can you add the `  * @return uintptr_t ` stanza to this comment?

File: thread/win32/omrthreadnuma.c [hunk 5]
```diff
@@ -148,3 +148,16 @@ omrthread_numa_get_node_affinity(omrthread_t thread, uintptr_t *numaNodes, uintp
...
+	}  else {
+		return 0;
```
Comment (DanHeidinga) [inline thread/win32/omrthreadnuma.c]: Should be possible to do this with a single return statement similar to the linux version

**Comment by charliegracie** [comment]
The Windows compile failure looks real.

**Comment by charliegracie** [comment]
@genie-omr build win

**Comment by dnakamura** [comment]
yeah I believe the windows compile failure is just a matter of a missing include file. Hunting it down now

**Comment by dnakamura** [comment]
After some digging it seems the `GetCurrentProcessorNumberEx` function was introduced in win. And we currently target windows xp (`057c2f9ba8`). @samolisov Is there a reason you put back the preprocessor defines to target win XP?

**Comment by samolisov** [comment]
@dnakamura we had a conversation with @youngar related to PR #2434 Please have a look at the comment https://github.com/eclipse/omr/pull/2434#issuecomment-381816845

**Comment by dnakamura** [comment]
Opened PR to update minimum version to win 7 (#3445) since AFAIK it is the minimum supported windows  version

**Comment by charliegracie** [comment]
Both of the Windows builds failed.  Any thoughts @dnakamura ?

**Comment by dnakamura** [comment]
@charliegracie I believe the windows CI builds ran before #3445 landed.

Also I have added the aix support code, removing WIP

File: include_core/thread_api.h (hunk 4 above)
Comment (DanHeidinga) [reply include_core/thread_api.h]: @dnakamura Can you address this review comment?

File: thread/linux/omrthreadnuma.c (hunk 1 above)
Comment (DanHeidinga) [reply thread/linux/omrthreadnuma.c]: @dnakamura was this comment addressed somewhere else?

**Comment by DanHeidinga** [comment]
@dnakamura I took a look at this again and I don't think the review comments were addressed.  Can you update the change and I can take another look at it?

**Comment by dnakamura** [comment]
@DanHeidinga  should be resolved

**Comment by DanHeidinga** [comment]
@dnakamura I'm probably missing something obvious but it's not clear to me why the node is being incremented before returning it.  ie:
```c
    if (0 == syscall(SYS_getcpu, NULL, &node, NULL)) {
         ++node;
```
Can you clarify that?  (preferably with the comment in the code so we don't lose the explanation)

**Comment by dnakamura** [comment]
@DanHeidinga  The OS numbers numa nodes starting from 0, but in OMR, we use a value of 0 to indicate no associated numa node

**Comment by pshipton** [comment]
@charliegracie is there something else that needs to be addressed before this is ready to merge?

**Comment by charliegracie** [comment]
@genie-omr build all

**Comment by rwy0717** [comment]
@genie-omr build aarch64
rebuilding due to infrastructure issues

**Pull request merged**